libraries/USBSerial/src/USBSerial.h

```diff
@@ -0,0 +1,80 @@
+/******************************************************************************
+ * The MIT License
+ *
+ * Copyright (c) 2010 Perry Hung.
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *****************************************************************************/
+
+/**
+ * @brief Wirish virtual serial port
+ */
+
+ /*
+  * Arduino srl - www.arduino.org
+  * 2016 Jun 9: Edited Francesco Alessi (alfran) - francesco@arduino.org
+  */
+
+#ifndef _USB_SERIAL_H_
+#define _USB_SERIAL_H_
+
+#if !defined (USBCON)
+#warning "Using legacy CDC core (non pluggable)"
+#else
+#include "Stream.h"
+#include "usbd_core.h"
+
+/**
+ * @brief Virtual serial terminal.
+ */
+class USBSerial : public Stream {
+public:
+    USBSerial(void);
+
+    void begin(void);
+    void begin(int);
+    void end(void);
+
+    virtual int available(void);
+    virtual int peek(void);
+    virtual void flush(void);
+    virtual int read(void);
+
+    int availableForWrite(void);
+    virtual size_t write(uint8_t);
+    inline size_t write(unsigned long n) { return write((uint8_t)n); }
+    inline size_t write(long n) { return write((uint8_t)n); }
+    inline size_t write(unsigned int n) { return write((uint8_t)n); }
+    inline size_t write(int n) { return write((uint8_t)n); }
```

Comment:
Why all these `write` overloads? The official Arduino AVR core's CDC serial (and, AFAIK also other Stream subclasses) do not have these overloads: https://github.com/arduino/ArduinoCore-avr/blob/master/cores/arduino/USBAPI.h#L103-L104

Also, since integer types are implicitly convertable to uint8_t there is no need for them (as long as there is only single-argument version of write, which is how the Stream class was intended).

Reply:
Hi @matthijskooijman 
this was a legacy from https://github.com/arduino-org/arduino-core-stm32f4
https://github.com/arduino-org/arduino-core-stm32f4/blob/master/cores/arduino/usb_serial.h#L59

Reply:
Right,
in the new PR for USB, I'm inline with CDC form Arduino